core/src/main/java/com/datastax/oss/driver/internal/core/resolver/mockResolver/MockResolver.java

```diff
@@ -0,0 +1,83 @@
+package com.datastax.oss.driver.internal.core.resolver.mockResolver;
+
+import com.datastax.oss.driver.internal.core.resolver.Resolver;
+import com.datastax.oss.driver.internal.core.resolver.Response;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+/** Resolver implementation that allows returning custom results. */
+public class MockResolver implements Resolver {
+  private final MockResolverResultSource resultSource;
+
+  MockResolver(MockResolverResultSource resultSource) {
+    this.resultSource = resultSource;
+  }
+
+  /**
+   * Returns first {@link InetAddress} from predefined {@link Response} or forwards the call to
+   * {@link InetAddress#getByName(String)} if undefined.
+   *
+   * @param host host to resolve.
+   * @return address.
+   * @throws UnknownHostException
+   */
+  @Override
+  public InetAddress getByName(String host) throws UnknownHostException {
+    Response response = this.resultSource.getResponse(host);
+    if (response == null) {
+      return InetAddress.getByName(host);
+    }
+    return response.result()[0];
+  }
+
+  /**
+   * Returns all {@link InetAddress}es from predefined {@link Response} or forwards the call to
+   * {@link InetAddress#getAllByName(String)} if undefined.
```

Comment:
```suggestion
   * Returns overrided results for provided {@link host}, if one is present, otherwise the call to
   * Otherwise forwards the call to {@link InetAddress.getAllByName}
```